### PR TITLE
[Merged by Bors] - Add `--staking` flag

### LINKED ISSUE #1472: RFC: Add a --staking or --no-staking flag
## Overview

During the Medalla launch I noticed that many users didn't provide the `--eth1` or `--http` flags, meaning their validators were unable to function.

Here are two proposals to make it easier for stakers

## Proposal A: Staking by default

We could modify the `lighthouse bn` CLI flags as such:

- **Remove** `--http`.
- **Remove** `--eth1`.
- **Add** `--no-staking`: Disables eth1 connection and indicates to other block/attestation production components that they'll never be needed.
- **Add** `--no-http`: disables the http server.

So, we have the following commands:

- `lighthouse bn`
    - http: yes
    - eth1: yes
- `lighthouse bn --no-staking`
    - http: yes
    - eth1: no
- `lighthouse bn --no-staking --no-http`
    - http: no
    - eth1: no
- `lighthouse bn --no-http`
    - http: no
    - eth1: yes
    - *Note: I'm struggling to see a use-case for this but I also don't see harm in having it.*

## Proposal B: Optional staking

We could modify the `lighthouse bn` CLI flags as such:

- **Add** `--staking`: Is an explicit version of `--http --eth1`.

So, we have the following commands:

- `lighthouse bn`
    - http: no
    - eth1: no
- `lighthouse bn --staking`
    - http: yes
    - eth1: yes
- `lighthouse bn --eth1`
    - http: no
    - eth1: yes
- `lighthouse bn --eth1 --http`
    - http: yes
    - eth1: yes

*Thanks to @AgeManning for this proposal*

### PR DESCRIPTION
**Pull request by pawanjay176**
## Issue Addressed

Closes #1472 

## Proposed Changes

Add `--staking` ~~and`staking-with-eth1-endpoint`~~ flag to improve UX for stakers.

### CODE REVIEW

File: book/src/become-a-validator-source.md [hunk 1]
```diff
@@ -47,15 +47,22 @@ the internet and maintains a view of the chain.
...
 ```
 
 > The `--testnet` parameter is optional. Omitting it will default to the
 > current public testnet. Set the value to the testnet you wish to run on.
 > Current values are either `altona` or `medalla`. This is true for all the
 > following commands in this document.
 
->Note:  the `--http` flag enables the HTTP API for the validator client. And the `--eth1` flag tells the beacon node that it should sync with an Ethereum1 node (e.g. Geth). These flags are only required if you wish to run a validator.
+>Note:  the `--staking` flag is an alias for `--eth1 --http`. It is required only if you wish to run a staking validator.
+`--http` flag enables the HTTP API for the validator client. And the `--eth1` flag tells the beacon node that it should sync with an Ethereum1 node (e.g. Geth) on `http://localhost:8545`.
```
Comment (paulhauner) [inline book/src/become-a-validator-source.md:59]: I think these comments are no longer relevant.

File: beacon_node/src/cli.rs [hunk 2]
```diff
@@ -187,6 +187,29 @@ pub fn cli_app<'a, 'b>() -> App<'a, 'b> {
...
+            Arg::with_name("staking")
+                .long("staking")
+                .help("Standard option for a staking beacon node. Equivalent to \
+                `lighthouse bn --http --eth1 `. This will enable the http server on localhost:5052 \
+                and try connecting to an eth1 node on localhost:8545")
+                .takes_value(false)
+                .conflicts_with("staking-with-external-eth1-endpoint")
+        )
+        .arg(
+            Arg::with_name("staking-with-ext-eth1-endpoint")
```
Comment (paulhauner) [inline beacon_node/src/cli.rs:204]: I wondering if `--staking-with-ext-eth1-endpoint ...` is really easier than `--staking --eth1-endpoint ...`? Perhaps I'm missing something though?

**Review by paulhauner — CHANGES_REQUESTED** [review]
Nice, thanks for picking this up! I just have a question and an minor comment :)

File: beacon_node/src/cli.rs (hunk 2 above)
Comment (pawanjay176) [reply beacon_node/src/cli.rs:204]: My rationale for the `--staking-with-ext-eth1-endpoint` flag was that users might equate `--staking` with a sigp hosted eth1 node somehow (many seemed to think that's what `--eth1` did on discord). 
Although now that I think of it, `--staking` without the correct endpoint will fail loudly. Will remove `--staking-with..` and improve the error messages.

**Review by paulhauner — APPROVED** [review]
Great! I made some minor changes to the docs to fix a couple of typos, I hope you don't mind :)

bors r+

**Pull request closed without merging**